File: dotnet/Barchart.BinarySerializer/Buffers/Exceptions/InvalidByteArrayLengthException.cs
namespace Barchart.BinarySerializer.Buffers.Exceptions;

/// <summary>
///     Thrown when the length of a byte array is invalid.
/// </summary>
public class InvalidByteArrayLengthException : ArgumentOutOfRangeException
{
    #region Constructor(s)
    
    /// <summary>
    ///     Creates a new <see cref="InvalidByteArrayLengthException"/> instance.
    /// </summary>
    /// <param name="length">
    ///     The length of the byte array.
    /// </param>
    public InvalidByteArrayLengthException(int length) : base( $"The byte array length must be positive. The length was {length}.")
    {
        
    }
    
    #endregion
}

File: typescript/src/buffers/exceptions/invalid-byte-array-length-exception.ts
/**
 * Thrown when the length of a byte array is invalid.
 *
 * @public
 * @exported
 * @extends {RangeError}
 * @param {number} length - The length of the byte array.
 */
export class InvalidByteArrayLengthException extends RangeError {
    constructor(length: number) {
        super(`The byte array length must be a positive integer. The length was ${ length }.`);

        this.name = new.target.name;
    }
}
